### PR TITLE
Sahar/io buffer 20

### PR DESCRIPTION
**Pull request by sfatimar**
IO Buffer changed for OV API 2.0
Build Issues 

**Motivation and Context**
- New API 2.0 Introduced.. 
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/core/providers/openvino/ov_interface.h [hunk 1]
```diff
@@ -5,36 +5,52 @@
 
 #include <inference_engine.hpp>
 #if defined (OPENVINO_2022_1)
-#include <openvino/core/core.hpp>
-#include <openvino/runtime/runtime.hpp>
+#define OV_API_20
```
Comment (MaajidKhan) [inline onnxruntime/core/providers/openvino/ov_interface.h:8]: why do we need an additional macro defined (OV_API_20)?
We have OpenVINO_2022_1 macro defined right. why do we have to change this.

Comment (sfatimar) [reply onnxruntime/core/providers/openvino/ov_interface.h:8]: When OPENVINO_2022_2 comes in i will have to make changes in all files to adapt to New API. In this case I will just have to make changes in one place

File: cmake/onnxruntime_providers.cmake [hunk 2]
```diff
@@ -732,9 +733,9 @@ if (onnxruntime_USE_OPENVINO)
 
   if ((DEFINED ENV{OPENCL_LIBS}) AND (DEFINED ENV{OPENCL_INCS}))
     add_definitions(-DIO_BUFFER_ENABLED=1)
-    list(APPEND OPENVINO_LIB_LIST $ENV{OPENCL_LIBS} ${InferenceEngine_LIBRARIES} ${NGRAPH_LIBRARIES} ngraph::onnx_importer ${PYTHON_LIBRARIES})
+    list(APPEND OPENVINO_LIB_LIST $ENV{OPENCL_LIBS} openvino::frontend::onnx openvino::runtime ${InferenceEngine_LIBRARIES} ${NGRAPH_LIBRARIES} ngraph::onnx_importer ${PYTHON_LIBRARIES})
   else()
-    list(APPEND OPENVINO_LIB_LIST ${InferenceEngine_LIBRARIES} ${NGRAPH_LIBRARIES} ngraph::onnx_importer ${PYTHON_LIBRARIES})
+    list(APPEND OPENVINO_LIB_LIST openvino::frontend::onnx openvino::runtime ${InferenceEngine_LIBRARIES} ${NGRAPH_LIBRARIES} ngraph::onnx_importer ${PYTHON_LIBRARIES})
```
Comment (MaajidKhan) [inline cmake/onnxruntime_providers.cmake:738]: with 2022.1, the ${NGRAPH_LIBRARIES} ngraph::onnx_importer won't be part of OPENVINO_LIB_LIST. should we try to add different checks for openvino version and only add the appropriate dependencies else it is fine anyways?

Comment (sfatimar) [reply cmake/onnxruntime_providers.cmake:738]: they are needed to maintain backward compatibility with 2021.4

File: onnxruntime/core/providers/openvino/ov_interface.h (hunk 1 above)
Comment (MaajidKhan) [reply onnxruntime/core/providers/openvino/ov_interface.h:8]: Yes makes sense.

File: cmake/onnxruntime_providers.cmake (hunk 2 above)
Comment (MaajidKhan) [reply cmake/onnxruntime_providers.cmake:738]: Right. I was thinking to have some cool cmake checks like:

if (OPENVINO_VERSION VERSION_LESS_EQUAL "2020.4")
    list(APPEND OPENVINO_LIB_LIST ${InferenceEngine_LIBRARIES} ${NGRAPH_LIBRARIES} ngraph::onnx_importer 
else()
   list(APPEND OPENVINO_LIB_LIST openvino::frontend::onnx openvino::runtime ${InferenceEngine_LIBRARIES} ${NGRAPH_LIBRARIES} ngraph::onnx_importer ${PYTHON_LIBRARIES})
  endif()

Comment (MaajidKhan) [reply cmake/onnxruntime_providers.cmake:738]: But I think whatever we have should be fine.

**Comment by MaajidKhan** [comment]
LGTM overall. @sfatimar  please go ahead and merge it in.

**Pull request merged**